ipu-plugin/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/intel/ipu-opi-plugins/ipu-plugin
 
-go 1.23.4
+go 1.23.6
```

Comment:
Let's not upgrade this. We need to move with dpu-operator. We just moved to 1.23.4 from 1.22.4 because they did so